indico/modules/rb_new/schemas.py

```diff
@@ -82,19 +84,43 @@ class Meta:
 
 
 class BlockedRoomSchema(mm.ModelSchema):
-    room = Nested(RoomSchema, many=False, only=('id', 'name', 'large_photo_url', 'full_name'))
+    room = Nested(RoomSchema, only=('id', 'name', 'sprite_position', 'full_name'))
 
     class Meta:
         model = BlockedRoom
         fields = ('rejection_reason', 'room')
 
 
+class BlockingPrincipalSchema(mm.ModelSchema):
+    identifier = Method('get_identifier')
+    full_name = String()
+    provider = String(missing=None)
+    email = String(missing=None)
+    is_group = Boolean(missing=False)
+
+    class Meta:
+        model = BlockingPrincipal
+        fields = ('id', 'identifier', 'name', 'is_group', 'email', 'full_name', 'provider')
+
+    def get_identifier(self, blocking_principal):
+        if blocking_principal.is_group:
+            return 'Group:{}:{}'.format(blocking_principal.provider or '',
```

Comment:
why not add a property for this on BlockingPrincipal?

or maybe a util function so you can use it both for the principal and the groupproxy itself?